src/Geometry/CubeGenerator.cpp

```diff
@@ -0,0 +1,1582 @@
+              //                                                              (((numCubesPerPart[2]-1)*numCubesPerPart[1]+yy)*numCubesPerPart[0]+xx)
+              //                                                              * 5 + 3;
+              //                                                              elemMPIIndices[(((numCubesPerPart[2]-1)*numCubesPerPart[1]+yy)*numCubesPerPart[0]+xx)
+              //                                                              * 20 + 15] =
+              //                                                              nextMPIIndex++;
+              //                                                      }
+            }
+          }
+
+          size_t start[3] = {(z * numPartitions[1] + y) * numPartitions[0] + x, bndSize, 0u};
+          size_t count[3] = {1, 1, static_cast<unsigned int>(nextMPIIndex)};
+          checkNcError(nc_put_var1_int(ncFile, ncVarBndElemSize, start, &nextMPIIndex));
+          int rank = (((z + 1) % numPartitions[2]) * numPartitions[1] + y) * numPartitions[0] + x;
+          rank = (rank + numPartitions[3]) % numPartitions[3];
+          checkNcError(nc_put_var1_int(ncFile, ncVarBndElemRank, start, &rank));
+          checkNcError(nc_put_vara_int(ncFile, ncVarBndElemLocalIds, start, count, bndLocalIds));
+
+          bndSize++;
+        }
+
+        size_t start[3] = {(z * numPartitions[1] + y) * numPartitions[0] + x, 0, 0};
+        size_t count[3] = {1, numElemPerPart[3], 4};
+        checkNcError(nc_put_vara_int(ncFile, ncVarElemMPIIndices, start, count, elemMPIIndices));
+
+        checkNcError(nc_put_var1_uint(ncFile, ncVarBndSize, &start[0], &bndSize));
+        writes_done++;
+        loadBar(writes_done, netcdf_writes);
+      }
+    }
+  }
+  delete[] elemMPIIndices;
+  delete[] bndLocalIds;
+
+  // Set material zone to 1
+  int* elemGroup = new int[numElemPerPart[3]];
+  std::fill(elemGroup, elemGroup + numElemPerPart[3], 1);
+  for (unsigned int x = 0; x < numPartitions[3]; x++) {
+    size_t start[2] = {x, 0};
+    size_t count[2] = {1, numElemPerPart[3]};
+    checkNcError(nc_put_vara_int(ncFile, ncVarElemGroup, start, count, elemGroup));
+  }
+  delete[] elemGroup;
+
+  // Vertices
+  std::map<int, CubeVertex> uniqueVertices;
+  transform(vertexMap.begin(),
+            vertexMap.end(),
+            inserter(uniqueVertices, uniqueVertices.begin()),
+            [](const auto& pair) { return std::make_pair(pair.second, pair.first); });
+  //&seissol::geometry::CubeGenerator::flip_pair<CubeVertex, int>);  //
+  //&<namespace>::<class>::<member_function> to pass pointer to member
+
+  int* vrtxSize = new int[numPartitions[3]];
+  std::fill(vrtxSize, vrtxSize + numPartitions[3], uniqueVertices.size());
+  checkNcError(nc_put_var_int(ncFile, ncVarVrtxSize, vrtxSize));
+  delete[] vrtxSize;
+  writes_done++;
+  loadBar(writes_done, netcdf_writes);
+
+  double* vrtxCoords = new double[uniqueVertices.size() * 3];
```

Comment:
vector? (though that's probably something more long-term) Same several times above.

Reply:
Since you noted that this change would be something more long-term, I'd delay it to another PR for the time being.